scrapy/pipelines/__init__.py

```diff
@@ -21,7 +21,8 @@ def _get_mwlist_from_settings(cls, settings):
                 category=ScrapyDeprecationWarning, stacklevel=1)
             # convert old ITEM_PIPELINE list to a dict with order 500
             item_pipelines = dict(zip(item_pipelines, range(500, 500+len(item_pipelines))))
-        return build_component_list(settings['ITEM_PIPELINES_BASE'], item_pipelines)
+            settings.set('ITEM_PIPELINES', item_pipelines)
```

Comment:
I recalled that ITEM_PIPELINES can't be a list with the changes in `SettingsAttribute.set()`, ITEM_PIPELINES has as value a BaseSettings instance, if a user tries to set a list here it's going to throw an error when trying to merge them into those BaseSettings.